circular structure introduced with 1.8.16 leading to 500 error

Within [etherpad-lite#5304](https://github.com/ether/etherpad-lite/pull/5304/files#diff-746de7116b3379f2e595c8f507d34333163e59e934fc271924a2d673ac972903R45) included in the 1.8.16 release there was the database object added to the pad properties leading to following error:
```
[2021-12-11 22:09:07.139] [ERROR] API - TypeError: Converting circular structure to JSON
    --> starting at object with constructor 'Timeout'
    |     property '_idlePrev' -> object with constructor 'TimersList'
    --- property '_idleNext' closes the circle
    at JSON.stringify (<anonymous>)
    at build (/opt/etherpad-lite/node_modules/ep_redis_publisher/index.js:59:15)
    at Object.exports.publish [as hook_fn] (/opt/etherpad-lite/node_modules/ep_redis_publisher/index.js:87:17)
    at callHookFnSync (/opt/etherpad-lite/src/static/js/pluginfw/hooks.js:122:16)
    at /opt/etherpad-lite/src/static/js/pluginfw/hooks.js:195:54
    at Array.map (<anonymous>)
    at Object.exports.callAll (/opt/etherpad-lite/src/static/js/pluginfw/hooks.js:195:25)
    at Pad.appendRevision (/opt/etherpad-lite/src/node/db/Pad.js:116:11)
    at Pad.init (/opt/etherpad-lite/src/node/db/Pad.js:367:16)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
```


```diff
diff --git a/src/node/db/Pad.js b/src/node/db/Pad.js
index 08e9809582..1307bcb89a 100644
--- a/src/node/db/Pad.js
+++ b/src/node/db/Pad.js
@@ -21,7 +21,7 @@ const hooks = require('../../static/js/pluginfw/hooks');
 const promises = require('../utils/promises');
 
 // serialization/deserialization attributes
-const attributeBlackList = ['id'];
+const attributeBlackList = ['_db', 'id'];
 const jsonableList = ['pool'];
 
 /**
@@ -34,7 +34,15 @@ exports.cleanText = (txt) => txt.replace(/\r\n/g, '\n')
     .replace(/\t/g, '        ')
     .replace(/\xa0/g, ' ');
 
-const Pad = function (id) {
+const Pad = function (id, database = db) {
+  this._db = database;
   this.atext = Changeset.makeAText('\n');
   this.pool = new AttributePool();
   this.head = -1;
@@ -95,7 +103,7 @@ Pad.prototype.appendRevision = async function (aChangeset, author) {
   }
```

--- a/index.js
+++ b/index.js
@@ -3,6 +3,15 @@ const settings = require('ep_etherpad-lite/node/utils/Settings');
 
 const CHANNEL = settings.ep_redis_publisher.channel || 'from-etherpad-redis-channel';
 
+// Remove unnecessary data from message's body
+const sanitizeMessageBody = (body) => {
+  if (body.pad && body.pad._db) {
+    delete body.pad._db;
+  }
+  
+  return body;
+};
+
 // JSON.stringify filter to remove circular references
 // adapted from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Cyclic_object_value#circular_references
 const getCircularReplacer = () => {
@@ -74,7 +83,7 @@ const build = (name, body) => {
 
   return JSON.stringify({
     envelope: buildEnvelope(name),
-    core: buildCore(name, body),
+    core: buildCore(name, sanitizeMessageBody(body)),
   }, getCircularReplacer());
 };
 

--- a/index.js
+++ b/index.js
@@ -3,6 +3,22 @@ const settings = require('ep_etherpad-lite/node/utils/Settings');
 
 const CHANNEL = settings.ep_redis_publisher.channel || 'from-etherpad-redis-channel';
 
+// JSON.stringify filter to remove circular references
+// adapted from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Cyclic_object_value#circular_references
+const getCircularReplacer = () => {
+  const seen = new WeakSet();
+  
+  return (key, value) => {
+    if (typeof value === "object" && value !== null) {
+      if (seen.has(value)) return;
+      
+      seen.add(value);
+    }
+    
+    return value;
+  };
+};
+
 const retry = (options) => {
   if (options.error && options.error.code === 'ECONNREFUSED') {
     logger.error('redis', 'connection refused');
@@ -59,7 +75,7 @@ const build = (name, body) => {
   return JSON.stringify({
     envelope: buildEnvelope(name),
     core: buildCore(name, body),
-  });
+  }, getCircularReplacer());
 };
 
 exports.publish = (event, body) => {

